src/test/java/com/redhat/lightblue/camel/ProducerTestRoute.java

```diff
@@ -5,18 +5,22 @@
 import com.redhat.lightblue.camel.model.User;
 import com.redhat.lightblue.camel.request.LightblueInsertRequestFactory;
 import com.redhat.lightblue.camel.utils.JacksonXmlDataFormat;
-import com.redhat.lightblue.camel.utils.LightblueErrorVerifier;
+import com.redhat.lightblue.client.response.LightblueErrorResponseException;
 
-public class InboundTestRoute extends RouteBuilder {
+public class ProducerTestRoute extends RouteBuilder {
 
     @Override
     public void configure() throws Exception {
+
+        onException(LightblueErrorResponseException.class)
+            .bean(new FailureHandler())
+            .handled(true);
+
         from("direct:start")
-            .unmarshal(new JacksonXmlDataFormat(User[].class)).convertBodyTo(User[].class)
+            .unmarshal(new JacksonXmlDataFormat(User[].class))
+            .convertBodyTo(User[].class)
```

Comment:
I know this convertBodyTo call pre-existed your change, but I think it is actually unnessisary as the unmarshal call above would have done that. We could probably just remove it.